comm/src/main/scala/coop/rchain/comm/rp/Connect.scala

```diff
@@ -76,7 +76,9 @@ object Connect {
         local   <- RPConfAsk[F].reader(_.local)
         timeout <- RPConfAsk[F].reader(_.defaultTimeout)
         hb      = heartbeat(local)
+        timer   <- Metrics[F].startTimer("heartbeat")
         res     <- TransportLayer[F].roundTrip(peer, hb, timeout)
+        _       = timer.stop()
```

Comment:
This seems to be wrong. `Timer::stop` is an effectful value, in order to "apply" this effect you need to `flatMap` on it, like:
`_ <- timer.stop()`

Reply:
The latest commit makes this no longer effectful.